distribution/lib/Standard/Database/0.0.0-dev/src/Internal/Helpers.enso

```diff
@@ -32,7 +32,7 @@ check_integrity entity1 entity2 =
    - entity2: The entity to check against the first.
 check_connection : (Table | Column) -> (Table | Column) -> Boolean
 check_connection entity1 entity2 =
-    entity1.connection == entity2.connection
+    Meta.is_same_object entity1.connection entity2.connection
```

Comment:
@jdunkerley this is the result of our discussion with @Akirathan - turns out this place was the source of the problems (together with how old `==` short-circuited on `Meta.is_same_object` and the fact that `Function` equality currently (always?) returns `False`).

Do you think this change is OK? I think in principle we exactly want that operations can be performed only within the _same_ connection (not different connections with the same parameters for example). So this looks good to me.
I'm only worried about the case where we may create a copy of the connection if we change some parameters like changing current schema or similar. What do  you think?

Anyway, this seems to be in-line with the old behaviour anyway and makes all of our tests passing again - so I think good to keep it as-is for now. But I wanted to discuss if we think this is ok to stay in general or is it just a temporary workaround?

Reply:
This is a reasonable workaround for this.